torchx/lib/torchx/backend.ex

```diff
@@ -1081,27 +1081,102 @@ defmodule Torchx.Backend do
   end
 
   @impl true
-  def pad(out, tensor, constant, config) do
+  def pad(out, tensor, constant, input_config) do
     config =
-      config
+      input_config
       |> Enum.map(fn {a, b, c} ->
-        if a < 0 or b < 0 or c != 0 do
+        if c < 0 do
           raise ArgumentError, "{#{a}, #{b}, #{c}} padding is not supported"
         end
 
-        [a, b]
+        [max(a, 0), max(b, 0)]
       end)
       |> Enum.reverse()
       |> List.flatten()
 
     constant = Nx.to_number(constant)
 
     tensor
-    |> from_nx()
+    |> pad_internal(input_config)
+    |> slice_negative_padding(input_config)
     |> Torchx.pad(config, constant)
     |> to_nx(out)
   end
 
+  defp pad_internal(t, []), do: from_nx(t)
+
+  defp pad_internal(t, input_config) do
+    pad_sizes = Enum.map(input_config, &elem(&1, 2))
+
+    if Enum.all?(pad_sizes, &(&1 == 0)) do
+      from_nx(t)
+    else
+      extra_axes = t |> Nx.axes() |> Enum.map(fn axis -> 2 * axis + 1 end)
+
+      t_expanded = Enum.reduce(extra_axes, t, &Nx.new_axis(&2, &1))
```

Comment:
I just figured we can also do this:

```elixir
expanded_shape =
  shape_list
  |> Enum.flat_map(fn size -> [size, 1] end)
  |> List.to_tuple()
```

and then reshape. Probably gives a better sense of what the result is, pick whichever you like more :)

Reply:
That does seem cleaner!